timelock-impl/src/main/java/com/palantir/atlasdb/timelock/TimelockNamespaces.java

```diff
@@ -57,7 +57,8 @@ public final class TimelockNamespaces {
 
     private static final SafeLogger log = SafeLoggerFactory.get(TimelockNamespaces.class);
 
-    private final ConcurrentMap<String, TimeLockServices> services = new ConcurrentHashMap<>();
+    // increase initial capacity to reduce contention at startup
+    private final ConcurrentMap<String, TimeLockServices> services = new ConcurrentHashMap<>(1024);
```

Comment:
Can we put more technical details on why we want this in the comment? 

In addition to that, it would be useful to get an explanation of how we got to 1024 (in view for this to be changed in the future if the client count changes etc, client count -> capacity should be a good function to have)

Reply:
I added more comments and consolidated the logic into a new `ConcurrentMaps.newWithExpectedEntries(int)` method to make it clearer. Let me know if that makes sense or you want additional clarification added to the comments & JavaDoc